base/reduce.jl

```diff
@@ -721,4 +721,4 @@ julia> countnz(A)
 6
 ```
 """
-countnz(a) = count(x -> x != 0, a)
+countnz(a) = count(!iszero, a)
```

Comment:
The underlying operation this breaks in the #19561/#19589 tests is `countnz(["A" 0; 0 "B"])` which I think is a valid operation and should return 2, not error like this will now do.